### PR TITLE
#14 check response status

### LINKED ISSUE #14: Response status not checked when downloading the Server jar
When the Go client downloads the Server jar from GitHub, it does not verify the response status. When the file does not exist, the Go client ignores the response code 404, writes an HTML error page to the jar and tries to execute it which fails.

Instead the Go client must fail with an error message if the status code is not 2xx.

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #14

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,8 @@
+*.sh    text eol=lf
+*.bat   text eol=crlf
+
+server/pk_generated_parent.pom       linguist-generated=true
+server/versionsMavenPluginRules.xml  linguist-generated=true
+dependencies.md                      linguist-generated=true
+doc/changes/changelog.md             linguist-generated=true
+go-client/go.sum                     linguist-generated=true
```
Comment (redcatbear) [inline .gitattributes:8]: :+1:

File: go-client/version_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package exasol_test_setup_abstraction_go
```
Comment (redcatbear) [inline go-client/version_test.go:1]: A little documentation that this is a check for the project configuration would be nice.

File: go-client/version_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+func (suite *VersionSuite) readProjectKeeperConf() *projectKeeperConfig {
+	yamlFile, err := os.ReadFile("../.project-keeper.yml")
+	if err != nil {
+		suite.FailNowf("failed to read project keeper file: %v", err.Error())
+	}
+	config := &projectKeeperConfig{}
+	err = yaml.Unmarshal(yamlFile, config)
+	if err != nil {
+		suite.FailNowf("unmarshal failed: %v", err.Error())
```
Comment (redcatbear) [inline go-client/version_test.go:49]:
```suggestion
		suite.FailNowf("Failed to parse YAML in project keeper configuration: %v", err.Error())
```

File: go-client/download_test.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+				t.Errorf("expected error %q but got %q", test.expectedError, err.Error())
+			}
+		})
+	}
+}
+
+func TestDownloadSucceeds(t *testing.T) {
+	dir := t.TempDir()
+	localPath := path.Join(dir, "file")
+	err := downloadFile("https://example.com", localPath)
```
Comment (redcatbear) [inline go-client/download_test.go:36]: Relying on a 3rd-party website in a test. :thinking:

**Review by redcatbear — APPROVED** [review]

File: go-client/version_test.go (hunk 2 above)
Comment (kaklakariada) [reply go-client/version_test.go:1]: ✅

File: go-client/download_test.go (hunk 4 above)
Comment (kaklakariada) [reply go-client/download_test.go:36]: Better than having no test at all. This is a task for future me.

**Review by redcatbear — APPROVED** [review]

**Pull request merged**